lib/netext/httpext/request.go

```diff
@@ -322,84 +409,24 @@ func MakeRequest(ctx context.Context, preq *ParsedHTTPRequest) (*Response, error
 		}
 
 		if res.StatusCode == http.StatusUnauthorized {
-			body := ""
-			if b, err := ioutil.ReadAll(res.Body); err == nil {
-				body = string(b)
-			}
-
 			challenge := digest.GetChallengeFromHeader(&res.Header)
-			challenge.ComputeResponse(preq.Req.Method, preq.Req.URL.RequestURI(), body, username, password)
+			challenge.ComputeResponse(preq.Req.Method, preq.Req.URL.RequestURI(), body.(string), username, password)
 			authorization := challenge.ToAuthorizationStr()
 			preq.Req.Header.Set(digest.KEY_AUTHORIZATION, authorization)
 		}
+		// TODO: fix - this is very broken! we're always making 2 HTTP requests
+		// when digest authentication is enabled...
```

Comment:
Maybe link the issue (for future PRs)? I prefer if I know that there is already an issue to be able to look at and although I know the particular issue, someone else might not be aware that there is already an ongoing discussion. Also combining the two TODOs is probably good idea as they are very likely to be fixed at the same time and are for the same thing.